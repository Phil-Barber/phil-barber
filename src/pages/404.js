import React from 'react';
import styled from 'styled-components';
import Layout from '../components/layout';

const Container = styled.div`
  margin: 0 auto;
  max-width: 750px;
  padding-top: ${({ theme }) => theme.spacing.large};
`;

const Content = styled.div`
  font-size: 40px;
  font-weight: 600;
  height: 100vh;
`;

const HomeLink = styled.a`
  margin-top: ${({ theme }) => theme.spacing.large};
  text-decoration: none;
  color: ${({ theme }) => theme.colors.primary};
  font-size: 20px;
  font-weight: normal;
`;

const Back = styled.span`
  padding-left: ${({ theme }) => theme.spacing.small};
  font-size: 25px;
`;

const FourOhFour = () => (
  <Layout>
    <Container>
      <Content>
        <div>Â¯\_(ãƒ„)_/Â¯</div>
        <div>Nothing to see here folks!</div>
        <HomeLink href="/">
          Take me back
          {/* eslint-disable-next-line jsx-a11y/accessible-emoji */}
          <Back role="img" aria-label="back">
            ðŸ”™
          </Back>
        </HomeLink>
      </Content>
    </Container>
  </Layout>
);

export default FourOhFour;
