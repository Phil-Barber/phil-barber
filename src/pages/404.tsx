import * as React from 'react';
import styled from 'styled-components';
import PageWrapper from '../components/pageWrapper';
import { PageLayout } from '../components/styled';

const Content = styled.div`
  font-size: 40px;
  font-weight: 600;
  min-height: 100vh;
`;

const HomeLink = styled.a`
  margin-top: ${({ theme }) => theme.spacing.large};
  text-decoration: none;
  color: ${({ theme }) => theme.colors.primary};
  font-size: 20px;
  font-weight: normal;
`;

const Back = styled.span`
  padding-left: ${({ theme }) => theme.spacing.small};
  font-size: 25px;
`;

export const FourOhFour: React.FC<null> = () => (
  <PageWrapper>
    <PageLayout>
      <Content>
        <div>Â¯\_(ãƒ„)_/Â¯</div>
        <div>Nothing to see here folks!</div>
        <HomeLink href="/">
          Take me back
          {/* eslint-disable-next-line jsx-a11y/accessible-emoji */}
          <Back role="img" aria-label="back">
            ðŸ”™
          </Back>
        </HomeLink>
      </Content>
    </PageLayout>
  </PageWrapper>
);

export default FourOhFour;
