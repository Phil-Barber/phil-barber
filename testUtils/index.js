import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../src/style';
import '@testing-library/jest-dom/extend-expect';

const customRender = (ui, options, store) => {
  const AllTheProviders = ({ children }) => (
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
  );
  return render(ui, { wrapper: AllTheProviders, ...options });
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
