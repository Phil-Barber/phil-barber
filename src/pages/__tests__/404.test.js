import React from 'react';
import { render } from '../../../testUtils';
import { FourOhFour } from '../404';

describe('FourOhFour', () => {
  it('Matches snapshot', () => {
    const { container } = render(<FourOhFour />);
    expect(container).toMatchSnapshot();
  });
});
