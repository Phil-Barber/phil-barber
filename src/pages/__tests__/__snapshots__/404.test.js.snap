// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FourOhFour Matches snapshot 1`] = `
<div>
  <div
    class="headroom-wrapper"
  >
    <div
      class="headroom headroom--unfixed"
      style="position: relative; top: 0px; left: 0px; right: 0px; z-index: 1; transform: translate3D(0, 0, 0);"
    >
      <div
        class="sc-AxjAm fEzfgO"
      >
        <a
          class="sc-AxhUy bILkkK"
          href="/"
        >
          <h1>
            Test title
          </h1>
        </a>
        <span
          class="sc-AxirZ gtVPNy"
          style="transform: rotate(0deg);"
        >
          &gt;
        </span>
      </div>
    </div>
  </div>
  <div
    class="sc-AxiKw jmqPVm"
  >
    <div
      class="sc-AxheI hhwjIJ"
    >
      <div
        class="sc-AxmLO jVuMPO"
      >
        <div
          class="sc-fzozJi la-dIuB"
        >
          <div>
            Â¯\\_(ãƒ„)_/Â¯
          </div>
          <div>
            Nothing to see here folks!
          </div>
          <a
            class="sc-fzoLsD tymYp"
            href="/"
          >
            Take me back
            <span
              aria-label="back"
              class="sc-fzpans eQLfzf"
              role="img"
            >
              ðŸ”™
            </span>
          </a>
        </div>
      </div>
    </div>
    <div
      class="sc-AxhCb bdrWTY"
    />
  </div>
</div>
`;
